### ISSUE TITLE
ESP8266 only stable for a week

### ISSUE THREAD

**Issue by AIXruth** [madmartin/Jarolift_MQTT#54]
My ESP8266 is hanging after round about a week. I couldn't find the reason yet. 
Will a second remote controller in use negativ influence the device counter?

**Comment by HerrRiebmann** [comment]
Does the ESP hung up or does the shutter stop to recognize the ESP?

I have two ESP32´s because the shutters were too far away (different story). After a while, some shutter did not work properly.
Since I know they all share one counter and listen to the signals, even if they were not controlled, I changed the "channel serial number prefix". Seperated the devices on the ESP´s and learned them new on the shutters.

And still, the shutter, which will listen to the commands from the "wrong" device, will stop working after a while. I have to re-learn it or just wait for about 10 minutes to be able to control it again.

The WAF is quite bad with a sometimes working solution like that :(
Two seperate devices, different serial number and counter. No channel overlaps.
Additional remote controller currently not in use...

**Comment by AIXruth** [comment]
After a few days the ESP8266 has no control over any shutter anymore. The shutter-buttons are still working and logging, but the shutters itself doesn't react. Using the "save" or "reboot" button on the system page results in a hanging ESP. The IP doesn't respond anymore till I reset the device using the power switch. Afterwards I have to re-learn the shutters again.

**Comment by HerrRiebmann** [comment]
On a classic Arduino I´d say you have an EEPROM issue. But the ESP doesn´t have one!
I´d try to grab the latest ESP Core and Libraries, to ensure it´s not a bug.
If this does not work fine, I´d get an ESP32 and give it a try. For 5-6 bucks, no big deal.

At least, my ESP32´s were running for months staight. Only one or two shutters go crazy.

**Comment by johi-1** [comment]
Do I have to change anything before flashing an ESP32 with your software?

**Comment by HerrRiebmann** [comment]
No, you just need to flash it onto it with the ESP right board settings.

**Comment by atze09** [comment]
> No, you just need to flash it onto it with the ESP right board settings.

Would it be also possible to flash it on a esp with ethernet? 
something similar to this?
https://www.roboter-bausatz.de/p/wt32-eth01-esp32-modul-mit-ethernet-bluetooth-wifi

I have a lot of wifi issues.

**Comment by HerrRiebmann** [comment]
@atze09 I think so, but I cannot tell you how much code you need to change.
Or if the firmware "rerouts" the WiFi traffic automatically to the ethernet. I have not worked with that version until yet.

**Comment by stalakerob** [comment]
@atze09 I tried to flash in on a ESP32 but that didn't work. I tried to adapt the code and libraries for the ESP32 but I gave up. My coding knowledge is too limited to do that. 

PS The problem most likely is based on connectivity issues between ESP and Jarolift. If too many commands are missed than the counter is out of sync and you need to re-learn your esp to Jarolift. I had that too. Then I changed the position of the ESP and since then it works without any issues.

**Comment by AIXruth** [comment]
@atze09: The current code is based on a WIFI connection. If you re-program the network part, it will be possible.

Based on my first post, I have created the following workaround that works for me:
I switch on the ESP via a switch socket, switch the shutters twice at intervals and then switch the ESP off again.
And yes, a more reliable ESP would also do the job without switch socket.

**Comment by atze09** [comment]
Migration to esp32 eth01 or in general to esphome project world be awesome.
My coding skills are definitely to low for according this.
